### PR TITLE
[Feature] Add operation loop

### PR DESCRIPTION
**Pull request by KaterinaWire**
## What's new in this PR?

1. Added `OperationLoop`.
2. Added `PushNotificationStrategy`

### CODE REVIEW

File: Sources/SharingSession.swift [hunk 1]
```diff
@@ -0,0 +1,358 @@
...
+}
+
+/// A Wire session to share content from a share extension
+/// - note: this is the entry point of this framework. Users of
+/// the framework should create an instance as soon as possible in
+/// the lifetime of the extension, and hold on to that session
+/// for the entire lifetime.
+/// - warning: creating multiple sessions in the same process
+/// is not supported and will result in undefined behaviour
+public class SharingSession {
```
Comment (typfel) [inline Sources/SharingSession.swift:164]: Let's rename to `NotificationSession`?

Comment (typfel) [reply Sources/SharingSession.swift:164]: And update the description

File: Sources/SharingSession.swift [hunk 2]
```diff
@@ -0,0 +1,358 @@
...
+        let requestGeneratorStore = RequestGeneratorStore(strategies: strategyFactory.strategies)
+        
+        let operationLoop = RequestGeneratingOperationLoop(
+            userContext: contextDirectory.uiContext,
+            syncContext: contextDirectory.syncContext,
+            callBackQueue: .main,
+            requestGeneratorStore: requestGeneratorStore,
+            transportSession: transportSession
+        )
+        print("<<< OperationLoop was created \(operationLoop)")
```
Comment (typfel) [inline Sources/SharingSession.swift:305]: Delete debug code?

File: Sources/Synchronization/OperationLoop.swift [hunk 3]
```diff
@@ -0,0 +1,242 @@
...
+
+import Foundation
+import CoreData
+import WireTransport
+import WireRequestStrategy
+
+let contextWasMergedNotification = Notification.Name("zm_contextWasSaved")
+
+
+final class RequestGeneratorStore {
```
Comment (typfel) [inline Sources/Synchronization/OperationLoop.swift:27]: This file copied unmodified from `wire-ios-sharing-session`? We could then move It to `wire-ios-request-strategy`

File: Sources/SharingSession.swift (hunk 2 above)
Comment (KaterinaWire) [reply Sources/SharingSession.swift]: Done

File: Sources/Synchronization/OperationLoop.swift (hunk 3 above)
Comment (KaterinaWire) [reply Sources/Synchronization/OperationLoop.swift:27]: Done

File: Sources/SharingSession.swift (hunk 1 above)
Comment (KaterinaWire) [reply Sources/SharingSession.swift:164]: Done

File: Sources/Synchronization/OperationLoop.swift (hunk 3 above)
Comment (typfel) [reply Sources/Synchronization/OperationLoop.swift]: Sorry maybe was unclear but I was referring to all classes/structs inside this source file, not only the `RequestGeneratorStore`.

File: Resources/Info.plist [hunk 4]
```diff
@@ -18,6 +18,11 @@
 	<string>$(MARKETING_VERSION)</string>
 	<key>CFBundleSignature</key>
 	<string>????</string>
+	<key>NSAppTransportSecurity</key>
```
Comment (typfel) [inline Resources/Info.plist:21]: We should probably revert this since we only plan on talking to our own server.

Comment (KaterinaWire) [reply Resources/Info.plist]: Reverted

File: Sources/Synchronization/OperationLoop.swift (hunk 3 above)
Comment (KaterinaWire) [reply Sources/Synchronization/OperationLoop.swift]: During a call we decided, for now, to keep Operation loop in NE.

**Pull request merged**